gateway-core/components/org.wso2.carbon.gateway.core/src/test/java/org/wso2/carbon/gateway/core/config/WUMLListenerTest.java

```diff
@@ -41,17 +46,108 @@ public void setup() {
         OutboundEPProviderRegistry.getInstance().registerOutboundEPProvider(new TestOutboundEPProvider());
     }
+
     @Ignore
     @Test()
     public void testListernerConstantsIflow() {
         Assert.assertTrue((parseIflow("/integration-flows/constants.iflow")));
     }
+
     @Ignore
     @Test
     public void testListernerVariableAssignmentIflow() {
         Assert.assertTrue((parseIflow("/integration-flows/variable-assignment.iflow")));
     }
 
+    @Test
+    public void testPassThroughObjectModelCreation() {
+        String iFlowResource = "/integration-flows/passthrough.xyz";
+        String integrationName = "passthrough";
+        String resourceName = "passthrough";
+        Boolean asExpected;
+
+        Assert.assertTrue("Error while deploying config :" + iFlowResource, deployArtifacts(iFlowResource));
+
+        Integration sampleIntegration = IntegrationConfigRegistry.getInstance().getIntegrationConfig(integrationName);
+        MediatorCollection generatedMediatorCollection = sampleIntegration.getResource(resourceName).getDefaultWorker()
+                .getMediators();
+        List<Mediator> mediatorList = generatedMediatorCollection.getMediators();
+
+        // Evaluate the generated mediator collection
+        if (mediatorList.size() == 3) {
+            asExpected = (generatedMediatorCollection.getFirstMediator() instanceof PropertyMediator);
+            asExpected = (mediatorList.get(1) instanceof CallMediator) && asExpected;
+            asExpected = (mediatorList.get(2) instanceof RespondMediator) && asExpected;
+        } else {
+            asExpected = false;
+        }
+
+        // remove the created Integration
+        IntegrationConfigRegistry.getInstance().removeIntegrationConfig(sampleIntegration);
+
+        Assert.assertTrue("Object generation according to the provided NEL configuration has failed.", asExpected);
+    }
+
+    @Test
+    public void testFilterMediatorObjectCreation() {
+        String iFlowResource = "/integration-flows/filter.xyz";
+        String integrationName = "filter";
+        String resourceName = "passthrough";
+        Boolean asExpected;
+
+        Assert.assertTrue("Error while deploying config :" + iFlowResource, deployArtifacts(iFlowResource));
+
+        Integration sampleIntegration = IntegrationConfigRegistry.getInstance().getIntegrationConfig(integrationName);
+        MediatorCollection generatedMediatorCollection = sampleIntegration.getResource(resourceName).getDefaultWorker()
+                .getMediators();
+        List<Mediator> mediatorList = generatedMediatorCollection.getMediators();
+
+        // Evaluate the generated mediator collection
+        if (mediatorList.size() == 6) {
+            asExpected = (generatedMediatorCollection.getFirstMediator() instanceof PropertyMediator);
+            asExpected = (mediatorList.get(1) instanceof LogMediator) && asExpected;
```

Comment:
My position is not to combine expected behaviour of the test case into single boolean; makes you scratch your head in the middle of the night when the build failed.